### PR TITLE
feature(chutneyEnvironment): add condition to prevent that 2 targets have the same name in a ChutneyEnvironment

### LINKED ISSUE #38: 🚀| Warn when targets in the same environment have the same name
Since the target name is used as an identifier, having targets with the same name in the same environment is not recommended.
We could warn users and even prevent execution, since it could lead to unexpected system state.

Solution : 
In [Environments.kt](https://github.com/chutney-testing/chutney-kotlin-dsl/blob/4ae13611e431b7888bba2c040342ad33076405a7/example/src/main/kotlin/com/chutneytesting/Environments.kt) constructor, raise an exception if there are targets with same name

### PR DESCRIPTION
**Pull request by DelaunayAlex**
add condition to prevent that 2 targets have the same name in a ChutneyEnvironment

#### Issue Number
fixes #38 

#### Describe the changes you've made

Add init() in ChutneyEnvironment, check if 2 targets have the same name, and if so, throw an exception

#### Describe if there is any unusual behaviour of your code 
NA

#### Checklist

- [x] Refer to issue(s) the PR solves
- [x] New kotlin code is covered by tests
- [x] Add screenshots or gifs of the new behavior, if applicable.
- [x] All new and existing tests pass
- [x] No git conflict

### CODE REVIEW

File: dsl/src/test/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDslTest.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class ChutneyEnvironmentDslTest {
+
+    @Test
+    fun `Test duplicate taget name in environment`() {
+        val targets: List<ChutneyTarget> = listOf(
+            ChutneyTarget("toto", "url1"),
```
Comment (bessonm) [inline dsl/src/test/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDslTest.kt:11]: Can you add a second non unique target, and at least one unique ?
ChutneyTarget("bou","url3")
ChutneyTarget("bou","url4")
ChutneyTarget("unique","url")

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDsl.kt [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+    init {
+        val notUniqueTargets = targets.groupBy { it.name }
+            .entries
+            .stream()
+            .filter { entry -> entry.value.size > 1 }
+            .map<Any> { entry -> entry.key }
+            .collect(Collectors.toSet())
+        if (notUniqueTargets.isNotEmpty()) {
+            throw IllegalArgumentException("Targets are not unique : " + notUniqueTargets.joinToString(", "))
+        }
```
Comment (nbrouand) [inline dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDsl.kt:20]: you can avoid java api stream :
```
 targets.groupBy { it.name }
            .filterValues { it.size > 1 }
            .keys
```

File: dsl/src/test/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDslTest.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            ChutneyTarget("toto","url2")
+        )
+
+        assertThrows<IllegalArgumentException> {
+            ChutneyEnvironment("default value", "description", targets)
+        }
```
Comment (bessonm) [inline dsl/src/test/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDslTest.kt:17]: Assert that the error message contains both 'toto' & 'bou' as non unique targets and not 'unique'

**Review by bessonm — CHANGES_REQUESTED** [review]

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDsl.kt (hunk 2 above)
Comment (DelaunayAlex) [reply dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDsl.kt:18]: Fixed

File: dsl/src/test/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDslTest.kt (hunk 1 above)
Comment (DelaunayAlex) [reply dsl/src/test/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDslTest.kt:31]: Fixed

File: dsl/src/test/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDslTest.kt (hunk 3 above)
Comment (DelaunayAlex) [reply dsl/src/test/kotlin/com/chutneytesting/kotlin/dsl/ChutneyEnvironmentDslTest.kt]: Fixed

**Review by nbrouand — APPROVED** [review]

**Pull request merged**